pubnub-kotlin/pubnub-kotlin-api/src/iosMain/kotlin/com/pubnub/kmp/Converters.kt

```diff
@@ -1,10 +1,113 @@
 
 fun String.toNSData(): NSData? {
     return (this as NSString).dataUsingEncoding(NSUTF8StringEncoding)
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun createPubNubHashedPage(from: PNPage?): PubNubHashedPageObjC {
+    return PubNubHashedPageObjC(
+        start = if (from is PNPage.PNNext) { from.pageHash } else { null },
+        end = if (from is PNPage.PNPrev) { from.pageHash } else { null },
+        totalCount = null
+    )
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun createPNUUIDMetadata(from: PubNubUUIDMetadataObjC): PNUUIDMetadata {
+    return PNUUIDMetadata(
+        id = from.id(),
+        name = from.name(),
+        externalId = from.externalId(),
+        profileUrl = from.profileUrl(),
+        email = from.email(),
+        custom = from.custom() as? Map<String, Any?>, // TODO: Verify
+        updated = from.updated(),
+        eTag = from.eTag(),
+        type = from.type(),
+        status = from.status()
+    )
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun createPNChannelMetadata(from: PubNubChannelMetadataObjC): PNChannelMetadata {
+    return PNChannelMetadata(
+        id = from.id(),
+        name = from.name(),
+        description = from.descr(),
+        custom = from.custom() as? Map<String, Any?>, // TODO: Verify
+        updated = from.updated(),
+        eTag = from.eTag(),
+        type = from.type(),
+        status = from.status()
+    )
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun createObjectSortProperties(from: Collection<PNSortKey<PNKey>>) : List<PubNubSortPropertyObjC> {
+    return from.map {
+        PubNubSortPropertyObjC(
+            key = it.key.fieldName,
+            direction = it.dir
+        )
+    }
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun createPNChannelMembership(from: PubNubMembershipMetadataObjC): PNChannelMembership {
+    return PNChannelMembership(
+        channel = PNChannelMetadata(
+            id = from.channel()?.id() ?: "",
+            name = from.channel()?.name() ?: "",
+            description = from.channel()?.descr() ?: "",
+            custom = from.channel()?.custom()?.asMap() as? Map<String, Any?>,
+            updated = from.channel()?.updated(),
+            eTag = from.channel()?.eTag(),
+            type = from.channel()?.type(),
+            status = from.channel()?.status()
+        ),
+        custom = from.custom() as Map<String, Any>,
+        updated = from.updated() ?: "",
+        eTag = from.eTag() ?: "",
+        status = from.status()
+    )
+}
+
+@OptIn(ExperimentalForeignApi::class)
+fun createPNMember(from: PubNubMembershipMetadataObjC): PNMember {
+    return PNMember(
+        uuid = from.uuid()?.let { createPNUUIDMetadata(from = it) },
+        custom = from.custom() as Map<String, Any>, // TODO: Check
+        updated = from.updated() ?: "",
+        eTag = from.eTag() ?: "",
+        status = from.status()
+    )
+}
+
+inline fun <reified T, U> filteredList(from: List<*>?, mapper: (T) -> U): Collection<U> {
```

Comment:
I'd probably make it an extension function so you can chain it easily:

```inline fun <reified T, U> List<*>?.filteredList(mapper: (T) -> U): Collection<U> {
    return this?.filterIsInstance<T>()?.map { mapper(it) } ?: emptyList()
}```

Reply:
also `map { mapper(it) }` is the same as `map(mapper)`

Reply:
Alright, I'm not sure if the name of the function I proposed is the best one. What I want to do is:

1. Ensuring that a collection contains `T` elements
2. If yes, transform each element into `U` by calling `((T) -> `U)`

What about naming it like `transform`?

Reply:
how about `filterAndMap` ?

Reply:
Sounds good, done!